### PR TITLE
Modified puppet module to include bool for using no-use_srv_records flag

### PR DESCRIPTION
**Pull request by brettmilford**
##### SUMMARY
This change adds an option to toggle the --no-use_srv_records flag in a puppet agent run command with the puppet module

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Puppet module no-use_srv_records flag

### CODE REVIEW

File: lib/ansible/modules/system/puppet.py
Comment (resmo) [inline lib/ansible/modules/system/puppet.py]: I would change it in a way to toggle either`--use_srv_records` or --no-use_srv_records` whether this value is true or false

~~~
 no_use_srv_records=dict(type='bool'),

File: lib/ansible/modules/system/puppet.py
Comment (resmo) [inline lib/ansible/modules/system/puppet.py]:
~~~python
        if p['use_srv_records'] is not None:
            if not p['use_srv_records']:
                cmd += " --no-use_srv_records"
            else:
                cmd += " --use_srv_records"
~~~

does this make sense?

File: lib/ansible/modules/system/puppet.py
Comment (resmo) [inline lib/ansible/modules/system/puppet.py]: use_srv_records:

**Review by resmo — COMMENTED** [review]
needs_info

File: lib/ansible/modules/system/puppet.py
Comment (brettmilford) [reply lib/ansible/modules/system/puppet.py]: It makes perfect sense, I did originally set out to do that, but couldn't find any documentation on the `--use_srv_recrods` flag.

File: lib/ansible/modules/system/puppet.py
Comment (resmo) [inline lib/ansible/modules/system/puppet.py]: this should be validaidating to non None:

~~~python
if p['use_srv_records'] is not None:
~~~

**Comment by brettmilford** [comment]
@resmo Is there some additional information or other action required to merge this PR?
Thanks.
bot_status

**Comment by brettmilford** [comment]
bot_status

**Review by resmo — APPROVED** [review]
shipit

**Comment by resmo** [comment]
Thanks for pinging, I adjusted the version_added to 2.9 and wait for the build before merge. Thanks again!

**Pull request merged**
